File: Core/Stateflows.Common/System/Events/BehaviorInstancesResponse.cs
using Stateflows.Common;
using System.Collections.Generic;

namespace Stateflows.System.Events
{
    public class BehaviorDescriptor
    {
        public BehaviorId Id { get; set; }

        public BehaviorStatus Status { get; set; }
    }

    public sealed class BehaviorInstancesResponse : Response
    {
        public IEnumerable<BehaviorDescriptor> Behaviors { get; set; }
    }
}

File: Transport/@stateflows/common/src/events/behavior-instances.response.ts
import { Response } from "./response";
import { BehaviorId } from "../ids/behavior.id";
import { BehaviorStatus } from "../enums/behavior-status";

export class BehaviorDescriptor
{
    id: BehaviorId;
    status: BehaviorStatus;
}

export class BehaviorInstancesResponse extends Response {
    constructor(
        public behaviors: Array<BehaviorDescriptor>,
    ) {
        super();
    }
}